src/Scriban/Runtime/ScriptObject.cs

```diff
@@ -86,6 +90,9 @@ public ScriptObject(int capacity, bool? autoImportStaticsFromThisType, IEquality
         void IDictionary.Add(object key, object value)
         {
             this.AssertNotReadOnly();
+#if NET
+            value = value is JsonElement json ? json.ToScriban() : value;
```

Comment:
Do we really need to support dynamic support? Wouldn't the requirement of using `ScriptObject.Import` be enough? It feels quite specific and it would open the door to `is Yaml`, `is Xml`...etc.

Reply:
In my latest commit, I removed all unnecessary JSON handling overloads and conditional statements. I retained only the elements necessary for the `TestModelWithJsonElement.cs` test cases, as I need this feature for my project where I utilize Scriban. I would really like to maintain this functionality. What are your thoughts?

Reply:
> I retained only the elements necessary for the TestModelWithJsonElement.cs test cases, as I need this feature for my project where I utilize Scriban.

I prefer that Scriban supports import/export of JSON to/from `ScriptObject`, but I really would like to avoid introducing a special case in the accessor of `ScriptObject`. The conversion of a `JsonElement` should only happen at import time, not implicitly at runtime on your behalf, otherwise it is making a significant exception. If we were starting to do that we would have to change also the `TypedObjectAccessor` and other accessors.

I don't think that it will change much `TestModelWithJsonElement.cs` as I believe import time should cover that. And if it wouldn't for your use case, you can always inherit from `ScriptObject` in your app and override set/get value if you really want to have such behavior for your app.

Reply:
So, if I understand you correct then removing every addition to ScriptObject except the `From` static method and implementing the JsonElement handling in the `Import` extension would be ok, right?

Reply:
Yep and the tests should still work